spec/migrations/20170927135007_migrate_dialog_field_associations_to_use_new_relationship_spec.rb

```diff
@@ -0,0 +1,63 @@
+require_migration
+
+describe MigrateDialogFieldAssociationsToUseNewRelationship do
+  let(:dialog_field_stub) { migration_stub(:DialogField) }
+  let(:dialog_group_stub) { migration_stub(:DialogGroup) }
+  let(:dialog_tab_stub) { migration_stub(:DialogTab) }
+  let(:dialog_stub) { migration_stub(:Dialog) }
+  let(:dialog_field_association_stub) { migration_stub(:DialogFieldAssociation) }
+
+  migration_context :up do
+    before(:each) do
+      dialog_group_stub.create!(:id => 2, :dialog_tab_id => 8, :position => 7)
+      dialog_tab_stub.create!(:id => 8, :dialog_id => 10, :position => 4)
+      dialog_stub.create!(:id => 10)
```

Comment:
Do not use manual ids because it bypasses proper region testing.

Reply:
Instead create the record, let ActiveRecord create a proper id, then use that id after the fact to update the associations.